File: src/main/java/xyz/avarel/waffles/nodes/builtin/AddBuiltinNode.java
package xyz.avarel.waffles.nodes.builtin;

import com.oracle.truffle.api.dsl.GenerateNodeFactory;
import com.oracle.truffle.api.dsl.Specialization;
import com.oracle.truffle.api.nodes.NodeInfo;
import xyz.avarel.waffles.nodes.BuiltinNode;

@GenerateNodeFactory
@NodeInfo(shortName = "+")
public abstract class AddBuiltinNode extends BuiltinNode {
    @Specialization
    public int add(int value0, int value1) {
        return value0 + value1;
    }
}

File: src/main/kotlin/xyz/avarel/waffles/nodes/builtin/AddBuiltinNode.kt
package xyz.avarel.waffles.nodes.builtin

import com.oracle.truffle.api.dsl.GenerateNodeFactory
import com.oracle.truffle.api.dsl.Specialization
import com.oracle.truffle.api.nodes.NodeInfo
import xyz.avarel.waffles.nodes.BuiltinNode

@GenerateNodeFactory
@NodeInfo(shortName = "+")
abstract class AddBuiltinNode : BuiltinNode() {
    @Specialization
    fun add(value0: Int, value1: Int) = value0 + value1
}
